platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/FeatureOverviewActivity.java

```diff
@@ -119,36 +123,42 @@ private void startFeature(Feature feature) {
   }
 
   private boolean requestLocationPermission(final int positionInList) {
-    if ((ContextCompat.checkSelfPermission(FeatureOverviewActivity.this,
-      Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED)
-      || (ContextCompat.checkSelfPermission(FeatureOverviewActivity.this,
-      Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED)) {
-      ActivityCompat.requestPermissions(FeatureOverviewActivity.this, new String[] {
-        Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION}, positionInList);
+    if(isRuntimePermissionsRequired()) {
+      locationActivityInList = positionInList;
+      permissionsManager.requestLocationPermissions(this);
       return true;
-    } else {
-      return false;
     }
+    return false;
+   }
+
+  @Override
+  public void onExplanationNeeded(List<String> list) {
+    Snackbar.make(
+      findViewById(android.R.id.content),
+      TextUtils.join("",list.toArray()),
+      Snackbar.LENGTH_SHORT).show();
   }
 
   @Override
-  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-    if (!isRuntimePermissionsRequired() || isPermissionAccepted(grantResults)) {
-      startFeature(features.get(requestCode));
-    } else {
+  public void onPermissionResult(boolean b) {
+    if(b){
+      startFeature(features.get(locationActivityInList));
+    }else{
       Snackbar.make(
         findViewById(android.R.id.content),
         "Can't open without accepting the location permission.",
         Snackbar.LENGTH_SHORT).show();
     }
   }
 
-  private boolean isRuntimePermissionsRequired() {
-    return android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M;
+  @Override
+  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+    super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+    permissionsManager.onRequestPermissionsResult(requestCode, permissions, grantResults);
   }
 
-  private boolean isPermissionAccepted(@NonNull int[] grantResults) {
-    return grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED;
+  private boolean isRuntimePermissionsRequired() {
```

Comment:
This has been ticketed out [here](https://github.com/mapbox/mapbox-java/issues/419#issuecomment-290442893), though the issue is still open.